mlrun/__main__.py

```diff
@@ -883,5 +887,26 @@ def dict_to_str(struct: dict):
     return ",".join(["{}={}".format(k, v) for k, v in struct.items()])
 
 
+def func_url_to_runtime(func_url):
+    try:
+        if func_url.startswith("db://"):
+            func_url = func_url[5:]
+            project, name, tag, hash_key = parse_function_uri(func_url)
+            mldb = get_run_db(mlconf.dbpath).connect()
+            runtime = mldb.get_function(name, project, tag, hash_key)
+        else:
+            func_url = "function.yaml" if func_url == "." else func_url
+            runtime = import_function_to_dict(func_url, {})
+    except Exception as e:
+        print("function {} not found, {}".format(func_url, e))
+        exit(1)
+
+    if not runtime:
+        print("function {} not found or is null".format(func_url))
+        exit(1)
+
+    return runtime
+
```

Comment:
@Hedingber raise is not good for CLI, u dont want to return a trace dump, rather a nice print, i can return `None` in the error cases and do the exit in the calling CLI. we cant place it in `helpers` since it will cause circular dependency  so i will keep it there for now.

Reply:
You're concern of trace dumps is correct, though this is not the right way of handling it, the code here is not simple short and clear, there is logic here, therefore like any other code it should just raise exceptions.
to handle not dumping trace dumps what should be done is simply wrap the function with one function the except everything and turn it into `exit(1)` - i.e. `exit` should be in place, not throughout all code.
anyways since all the file is currently written that way just leave as is, I'll refactor it later 🤷‍♂️